contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -165,13 +176,32 @@ abstract contract PoSValidatorManager is
         bool includeUptimeProof,
         uint32 messageIndex
     ) external {
+        uint256 reward = _initializeEndPoSValidation(validationID, includeUptimeProof, messageIndex);
+        if (reward == 0) {
+            revert ValidatorIneligibleForRewards();
+        }
+    }
+
+    function forceInitializeEndValidation(
+        bytes32 validationID,
+        bool includeUptimeProof,
+        uint32 messageIndex
+    ) external {
+        _initializeEndPoSValidation(validationID, includeUptimeProof, messageIndex);
+    }
+
+    function _initializeEndPoSValidation(
+        bytes32 validationID,
+        bool includeUptimeProof,
+        uint32 messageIndex
+    ) internal returns (uint256) {
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
 
         Validator memory validator = _initializeEndValidation(validationID);
 
         // Non-PoS validators are required to boostrap the network, but are not eligible for rewards.
         if (!_isPoSValidator(validationID)) {
-            return;
+            revert ValidatorNotPoS();
```

Comment:
I'm pretty sure this leaves no route for the initial validators to be removed from the validator set. The flow was the call to `_initializeEndValidation` above, and then returning so none of the rewards calculations or anything were dealt with.

Reply:
Good catch. This was causing the E2E tests to fail. Fixed now.